test/e2e/singlecluster/deployment_test.go

```diff
@@ -154,20 +152,17 @@ var _ = ginkgo.Describe("Deployment", func() {
 		pods := &corev1.PodList{}
 		gomega.Expect(k8sClient.List(ctx, pods, client.InNamespace(ns.Name),
 			client.MatchingLabels(deployment.Spec.Selector.MatchLabels))).To(gomega.Succeed())
+		gomega.Expect(pods.Items).To(gomega.HaveLen(3))
```

Comment:
Please describe (also in PR description) what was the problem and why the change helps.

Reply:
The problem is that sometimes workloads don’t have enough time to reach QuotaReservation=false (which we're checking below) after the pods start running. We should use `gomega.Eventually` to ensure this condition is met.